vcpkg/maintainers/authoring-tool-ports.md

```diff
@@ -0,0 +1,18 @@
+---
+title: Authoring Tool Ports
+description: Learn to create a Tool Port for usage in other ports.
+ms.date: 11/30/2022
+---
+# Authoring tool ports
+
+Ports providing a binary or tool for other ports are tool ports. For example the `vcpkg-tool-meson` port exposes the required meson python script to run meson based builds in a versioned manner. Tool ports are generally ment to replace `vcpkg_find_acquire_program` calls since the latter is not versioned and triggers an CI world rebuild if changed.
+
+Tool ports are implemented in different ways depending on their expected usage. 
+1. If a tool is expected to be provided in binary form and not ment for the user to be used the port should be named `vcpkg-tool-<toolname>`. The port should work like `vcpkg_find_acquire_program` and only download/provide the tool if another port requests its usage. This is done via `vcpkg-port-config.cmake` which has to included all logic to find or provide the tool. The artifacts of the tool have to be put into `${DOWNLOADS}/tools/<tool_name>-<tool_version>`. (This means that the tool port is actually just a [script port](./authoring-script-ports.md))
+2. If a tool is required to be build from source but is not to be used by the user the tools build artifacts should be stored into `${CURRENT_PACKAGES_DIR}/manual-tools/${PORT}`. Furthermore a `vcpkg-port-config.cmake` needs to be installed setting a variable to path of the relevant tool.
+3. If a tool is expected to be used outside of vcpkg and inside the users `CMakeLists.txt`. The tools build artifacts are installed to `${CURRENT_PACKAGES_DIR}/tools/${PORT}` and the port name should follow the normal vcpkg naming rules. A `vcpkg-port-config.cmake` is required to be installed like in case 2.
```

Comment:
Do you have an example and should `tools` features also install  `vcpkg-port-config.cmake` files

Reply:
this probably falls into case 3. I thought about dropping the `vcpkg-port-config.cmake` for that case unless it is expected to be referenced by a `portfile` directly. (or simply install an empty `vcpkg-port-config.cmake`).

Maybe it comes down to define a rule how to reference tool paths from portfiles in this case. Everything which is external to the port itself should be indirectly referenced via a `vcpkg-port-config.cmake` while a port referencing itself (e.g. code generators) is allowed to directly access it host build artifacts.

I can also imaging a `vcpkg_find_program` call which drops the acquire component but is aware of additional files being installed by ports.